### PR TITLE
Fix bug causing 100% CPU usage when starting Locust without interactive shell

### LINKED ISSUE #1629: Locust docker version 1.4.0 using 100% CPU on idle
Test environment:
Ubuntu 18.04.1
Docker version 19.03.6

Environment stays the same only the docker image version is changed.
Running the docker container with the same locustifle.py which is an example from https://hub.docker.com/r/locustio/locust

```python
from locust import HttpUser, task

class QuickstartUser(HttpUser):
    def on_start(self):
        self.client.post
("/login", json={"username":"foo", "password":"bar"})

    @task
    def hello_world(self):
        self.client.get("/hello")
        self.client.get("/world")

    @task(3)
    def view_item(self):
        for item_id in range(10):
            self.client.get(f"/item?id={item_id}", name="/item")
```

When running locustio/locust version 1.3.2 the idle CPU load is around 0.0% and stable. 

Running locustio/locust version 1.4.0 the idle CPU load goes up to 100.0% at once when the container is started and stays between 96.7% and 100.0%.

### PR DESCRIPTION
**Pull request by heyman**
Should fix #1629 .

### CODE REVIEW

File: locust/input_events.py [hunk 1]
```diff
@@ -94,6 +94,7 @@ def input_listener_func():
                     gevent.sleep(0.2)
         except Exception as e:
             logging.warning(f"Exception in keyboard input poller: {e}")
+            raise
```
Comment (heyman) [inline locust/input_events.py:97]: I've added this for a more robust code, but I'm not sure if there are any valid exceptions that could happen during a normal session. If so, we should probably handle them seprately.

**Comment by cyberw** [comment]
Cool stuff! Me and @DennisKrone have a fix on the way, but will integrate your check for isatty() into it (instead of relying on catching an exception, which was our first attempt)

**Comment by heyman** [comment]
> Cool stuff! Me and @DennisKrone have a fix on the way, but will integrate your check for isatty() into it (instead of relying on catching an exception, which was our first attempt)

Great!

**Comment by cyberw** [comment]
Superseded by #1631

**Comment by kurtmckee** [comment]
Looks like the `fix-1629` branch can be deleted.

**Comment by cyberw** [comment]
Thx!

**Pull request closed without merging**